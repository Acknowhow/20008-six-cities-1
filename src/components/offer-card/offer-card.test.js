import React from 'react';
import renderer from 'react-test-renderer';
import OfferCard from './offer-card.jsx';


const mock = {
  offer: {
    id: 5,
    [`is_premium`]: true,
    [`preview_image`]: `some/src`,
    price: 10012313,
    type: `apartment`,
    rating: 4.5,
    title: `Mock title`
  }
};

it(`OfferCard renders correctly`, () => {
  const handler = jest.fn();
  const {offer} = mock;
  const index = 2;

  const tree = renderer
    .create(<OfferCard
      offer={offer}
      mouseOverHandler={handler}
      index={index}
    />).toJSON();

  expect(tree).toMatchSnapshot();
});
