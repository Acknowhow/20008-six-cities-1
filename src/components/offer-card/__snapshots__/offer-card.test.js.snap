// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferCard renders correctly 1`] = `
<article
  className="cities__place-card place-card"
  data-index={2}
  onMouseEnter={[MockFunction]}
>
  <div
    className="place-card__mark"
  >
    <span>
      Downtown
    </span>
  </div>
  <div
    className="cities__image-wrapper place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height="200"
        src=""
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          17
        </b>
        <span
          className="place-card__price-text"
        >
          / 
          week
        </span>
      </div>
      <button
        className="place-card__bookmark-button button"
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "57.78%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        className="place-card__link"
        href="#"
      >
        Spartan place for saturday hangovers
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      Flat
    </p>
  </div>
</article>
`;
